Add random salting to the encryption

LucasRoesler opened the pull request:
- Add salting to the encryption
- Use absolute imports in the test app
- Update EncryptedChar unit test to test passing in a custom salt length
  to the field.

danaspiegel commented on cryptographic_fields/fields.py:
```diff
@@ -25,30 +29,37 @@
         self.unencrypted_max_length = self.max_length
-        self.max_length = calc_encrypted_length(self.unencrypted_max_length)
+        self.max_length = calc_encrypted_length(
+            self.unencrypted_max_length, self.salt_length)
```
@LucasRoesler this actually doesn't look quite right. You're passing in a salt_length (an number), but its being used in the ```calc_encrypted_length``` as the actual salt value.

danaspiegel commented on cryptographic_fields/fields.py:
```diff
@@ -25,30 +29,37 @@
 
 class EncryptedMixin(object):
     def __init__(self, *args, **kwargs):
+        self.salt_length = kwargs.pop('salt', 2)
```
@LucasRoesler The code leads to some confusion. We should either:

1. initialize with a ```salt_length```, then calculate it, or
2. initialize with a ```salt_value``` itself

Regardless, as a result of the code you've modified in this file, there's the confusion between a value and a length becomes apparent.

danaspiegel commented:
@LucasRoesler I think the biggest issue with your changes is the nomenclature you use. I think all the passed variables currently called ```salt``` should be called ```salt_length``` to make it clear that the actual generation of the randomized salted value happens at encryption time, and that the developer is really just specifying the length of the salted bytes that's generated.

Further, I have a question re: how useful it is to allow changing of the salt length on a per field basis. We don't want the salted value to be easily brute force guessed, but is 10 bytes much different from 12 or 15? Why not just chose a secure length and make that the default so everyone can have a reasonable level of security. 99%+ of developers aren't going to have an opinion on the length of salt used.

cc: @levigross

danaspiegel commented on cryptographic_fields/fields.py:
```diff
@@ -25,30 +29,37 @@
-def encrypt_str(s):
-    # be sure to encode the string to bytes
+def encrypt_str(s, salt):
+    s = u'{salt}:{string}'.format(
```
@LucasRoesler This is actually dangerous, since the ```:``` character might show up in the salted bytes. We should make sure to replace that character if it appears before generated the ```salt + string``` value that is encrypted.

LucasRoesler commented:
I made some tweaks to the salting stuff.  First, I made a poor naming decision, so I updated the parameter names to be clearer. Second, I changed how I was calculating the salt, so that I could more easily strip it out during the decrypt step.  Hopefully the code is clearer now.

LucasRoesler commented:
I am going to close this because I am silly and didn't read https://cryptography.io/en/latest/fernet/#implementation well enough, salting is already handled (properly I might add) by the cryptography library.
